app/models/grace_period_submission_rule.rb

```diff
@@ -25,6 +25,7 @@ def assignment_valid?
   end
 
   def apply_submission_rule(submission)
+    return submission if submission.revision_timestamp.nil?
```

Comment:
@david-yz-liu made a good point that we should be checking for `submission.is_empty` here instead. 

That will prevent us from applying a submission rule to students who simply create a grouping during the penalty period but still don't submit anything on time.